src/components/OrderBookWidget.tsx

```diff
@@ -224,103 +223,257 @@ function _printOrderBook(pricePoints: PricePointDetails[], baseToken: TokenDetai
   const { baseToken, quoteToken, networkId, hops } = props
+  const [chart, setChart] = useSafeState<null | am4charts.XYChart>(null)
+
   const mountPoint = useRef<HTMLDivElement>(null)
 
   useEffect(() => {
-    if (!mountPoint.current) return
-    const chart = draw(mountPoint.current, baseToken, quoteToken, networkId, hops)
+    if (!mountPoint.current) {
+      return
+    }
 
-    return (): void => chart.dispose()
-  }, [baseToken, quoteToken, networkId, hops])
+    const _chart = am4core.create(mountPoint.current, am4charts.XYChart)
 
-  return (
-    <Wrapper ref={mountPoint}>
-      Show order book for token {safeTokenName(baseToken)} ({baseToken.id}) and {safeTokenName(baseToken)} (
-      {quoteToken.id})
-    </Wrapper>
-  )
-}
+    // Create axes
+    const xAxis = _chart.xAxes.push(new am4charts.ValueAxis())
+    // Making the scale start with the first value, without empty spaces
+    // https://www.amcharts.com/docs/v4/reference/valueaxis/#strictMinMax_property
+    xAxis.strictMinMax = true
+    // How small we want the column separators be, in pixels
+    // https://www.amcharts.com/docs/v4/reference/axisrendererx/#minGridDistance_property
+    xAxis.renderer.minGridDistance = 40
+
+    // To start zoomed in when data loads
+    // https://www.amcharts.com/docs/v4/reference/valueaxis/#keepSelection_property
+    xAxis.keepSelection = true
+    // To allow we to zoom reaaaaly tiny fractions
+    xAxis.maxZoomFactor = 10 ** 18
+
+    const yAxis = _chart.yAxes.push(new am4charts.ValueAxis())
+
+    yAxis.keepSelection = true
+    yAxis.maxZoomFactor = 10 ** 18
+
+    // Colors
+    const colors = {
+      green: '#3d7542',
+      red: '#dc1235',
+    }
+
+    // Create series
+    const bidSeries = _chart.series.push(new am4charts.StepLineSeries())
+    bidSeries.dataFields.valueX = 'priceNumber'
+    bidSeries.dataFields.valueY = 'bidValueY'
+    bidSeries.strokeWidth = 1
+    bidSeries.stroke = am4core.color(colors.green)
+    bidSeries.fill = bidSeries.stroke
+    bidSeries.fillOpacity = 0.1
+
+    const askSeries = _chart.series.push(new am4charts.StepLineSeries())
+    askSeries.dataFields.valueX = 'priceNumber'
+    askSeries.dataFields.valueY = 'askValueY'
+    askSeries.strokeWidth = 1
+    askSeries.stroke = am4core.color(colors.red)
+    askSeries.fill = askSeries.stroke
+    askSeries.fillOpacity = 0.1
+
+    // Add cursor
+    _chart.cursor = new am4charts.XYCursor()
+
+    _chart.cursor.snapToSeries = [bidSeries, askSeries]
+
+    am4core.useTheme(am4themesSpiritedaway)
+    am4core.options.autoSetClassName = true
+
+    _chart.scrollbarX = new am4core.Scrollbar()
+    _chart.scrollbarY = new am4core.Scrollbar()
 
-export default OrderBookWidget
+    setChart(_chart)
+
+    return (): void => _chart.dispose()
+    // We'll create only one instance as long as the component is not unmounted
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [])
+
+  useEffect(() => {
+    if (chart) {
```

Comment:
nit: if we change this into 
```
if (!chart) {
   return
}
```
I understand it would be equivalent, and you reduce the cyclomatic complexity of this useEffect. Specially, if you also take my other nitpick (below)